### ISSUE TITLE
pretrain doesn't work on json\jsonl

### ISSUE THREAD

**Issue by SicariusSicariiStuff** [axolotl-ai-cloud/axolotl#1895]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

To work the same as when loading the dataset from HF

### Current behaviour

Asks for a custom .py script

### Steps to reproduce

Load a local json file:

pretraining_dataset: /home/sicarius/somefile.jsonl
    type: pretrain

### Config yaml

```yaml
pretraining_dataset: /home/sicarius/somefile.jsonl
    type: pretrain
```

### Possible solution

Treat it similarly as a loading a dataset from the HF hub

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

latest release

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Hey, sorry it's been a while. We are currently internally discussing providing better support for this and pre-training/sft in general. We plan to extend support to local and cloud storage (S3 etc).

**Comment by NanoCode012** [comment]
Hello, I missed updating this Issue. For anyone that stumbles upon this, the following now works on json/jsonl dataset (as well as csv, parquet, arrow, etc).

```yaml
pretraining_dataset:
  - path: json
    data_files:
      - A.jsonl
      - B.jsonl
      - C.jsonl
```


Ref: https://axolotl-ai-cloud.github.io/axolotl/docs/dataset-formats/#pre-training-from-local-dataset-files

---

Unfortunately, we haven't got to extend support to cloud storage yet.
